trin-core/src/content_key.rs

```diff
@@ -0,0 +1,57 @@
+use crate::portalnet::types::content_key::OverlayContentKey;
+use sha2::{Digest, Sha256};
+use ssz::{self, Decode, Encode};
+use ssz_derive::{Decode, Encode};
+
+//
+// *TODO*
+// This code has been duplicated from the trin-history crate to allow the trin-core crate to have
+// access to ContentKey derivation which enables `eth_getBlockByHash`. This is a temporary fix to
+// achieve a demo-able feature for Devconnect - April 2022. It's been decided by the team that this
+// is not a sustainable pattern, and a significant re-structuring of Trin is required, to be
+// evaluated post-Devconnect.
+//
```

Comment:
I agree that it is out of scope here, but I would prefer that would go ahead and move the code prior to the merge of this PR as opposed to doing the cleanup after. That reorganization PR would be simple to review and approve, so it wouldn't delay this in a significant way. It would also limit the scope of this PR to the new functionality that we want to add. I realize this may seem pedantic, but it feels like the right sequence of changes.